Code/Framework/AzFramework/AzFramework/Quality/QualityCVarGroup.cpp

```diff
@@ -0,0 +1,244 @@
+                return VisitResponse::Skip;
+            }
+
+            AZStd::string_view command = visitArgs.m_fieldName;
+            AZStd::string_view valueKey = visitArgs.m_jsonKeyPath;
+            if (visitArgs.m_type == Type::Array)
+            {
+                valueKey = FixedValueString::format("%.*s/%d", AZ_STRING_ARG(valueKey), qualityLevel);
+                if (visitArgs.m_registry.GetType(valueKey) == Type::NoType)
+                {
+                    // if the array index doesn't exist, use the value for the highest index
+                    ValueType maxQualityLevel = GetHighestQualityForSetting(visitArgs.m_jsonKeyPath);
+                    if (maxQualityLevel < 0)
+                    {
+                        AZ_Warning("QualityCVarGroup", false, "Not applying settings for %.s because the array is empty.", AZ_STRING_ARG(command));
+                        return VisitResponse::Continue;
+                    }
+                    valueKey = FixedValueString::format("%.*s/%d", AZ_STRING_ARG(visitArgs.m_jsonKeyPath), maxQualityLevel);
+                }
+            }
+
+            PerformConsoleCommand(command, valueKey);
+
+            return VisitResponse::Continue;
+        };
+        AZ::SettingsRegistryVisitorUtils::VisitObject(*m_settingsRegistry, settingsVisitorCallback, key);
+    }
+
+    void QualityCVarGroup::PerformConsoleCommand(AZStd::string_view command, AZStd::string_view key)
+    {
+        AZStd::string value;
+        switch (m_settingsRegistry->GetType(key))
+        {
+        case AZ::SettingsRegistryInterface::Type::FloatingPoint:
+            {
+                double floatingPointValue;
+                if (m_settingsRegistry->Get(floatingPointValue, key))
+                {
+                    value = AZStd::to_string(floatingPointValue);
+                }
+            }
+            break;
+        case AZ::SettingsRegistryInterface::Type::Boolean:
+            {
+                bool boolValue;
+                if (m_settingsRegistry->Get(boolValue, key))
+                {
+                    value = boolValue ? "True" : "False";
+                }
+            }
+            break;
+        case AZ::SettingsRegistryInterface::Type::Integer:
+            {
+                AZ::s64 intValue;
+                if (m_settingsRegistry->Get(intValue, key))
+                {
+                    value = AZStd::to_string(intValue);
+                }
+            }
+            break;
+        case AZ::SettingsRegistryInterface::Type::String:
+        default:
+            {
+                m_settingsRegistry->Get(value, key);
+            }
+            break;
+        }
+
+        if (!value.empty())
+        {
+            m_console->PerformCommand(command, { value }, AZ::ConsoleSilentMode::Silent);
+        }
+        else
+        {
+            AZ_Warning("QualityCVarGroup", false, "Failed to convert the value for %.*s to a string, or the value is empty.", AZ_STRING_ARG(command));

```

Comment:
I lean towards now, not issuing AZ_Warnings/AZ_Errors when possible to avoid polluting the log and console and instead return an AZ::Outcome/AZStd::expected where the error type is a string that contains the error text.

i.e
```c++
using PerformCommandOutcome = AZStd::expected<void, AZStd::string>;
PerformCommandOutcome QualityCVarGroup::PerformConsoleCommand(AZStd::string_view command, AZStd::string_view key)
{
     // ... existing logic
    PerformCommandOutcome resultOutcome;
    resultOutcome = AZStd::unexpected(AZStd::string::format("Failed to convert the value for %.*s to a string, or the value is empty.", AZ_STRING_ARG(command)));
}